### PR TITLE
Feature/collateral factory #94

### LINKED ISSUE #94: [MarketCollateralPool] Implement factory that could possibly create collateral pool and link to MarketContract
## Before you `start work`

Please read our contribution [guidelines](https://docs.marketprotocol.io/#contributing) and if there is a bounty involved  please also see [here](https://docs.marketprotocol.io/#gitcoin-and-bounties)

If you have ongoing work from other bounties with us where funding has not been released, please do not pick up a new issue.  We would like to involve as many contributors as possible and parallelize the work flow as much as possible.

Please make sure to comment in the issue here immediately after starting work so we know your plans for implementation and a timeline.

Please also note that in order for work to be accepted, all code must be accompanied by test cases as well.

### User Story
As a users, the less transactions I have to sign the better!

### Why Is this Needed?

*Summary*: Could possibly streamline our deployment process and ensure contracts are being created using known source code

### Description

*Type*: Feature

### Current Behavior
We have three transactions to full deploy a MARKET Protocol Smart Contract

### Expected Behavior
Can this be condensed into two, or perhaps even a single transaction?!

### PR DESCRIPTION
**Pull request by pelsasser**
##### Description
Creates a collateral pool factory that streamlines deployment, adds more security and reduces number of needed transactions for a user to deploy a contract (down to 2 vs 3) fixes #94 

Also fixes some style issues as well as the deprecate `keccak` calls without using `abi.encodePacked`

##### Checklist

- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [x] Tests coverage hasn't decreased

##### Affected core subsystem(s)
protocol, docs will need to be updated as well as dApp

##### Testing
tests updated and passing with new migration script.

##### Refers/Fixes
fixes #94

### CODE REVIEW

**Comment by pelsasser** [comment]
@eswarasai - once this is reviewed, I will work to create the first release of the new @marketprotocol/marketprotocol npm package

File: contracts/MarketContract.sol [hunk 1]
```diff
@@ -340,4 +345,10 @@ contract MarketContract is Creatable {
         settlementPrice = finalSettlementPrice;
         emit ContractSettled(finalSettlementPrice);
     }
+
+    ///@dev Throws if called by any account other than the factory.
```
Comment (cryptomental) [inline contracts/MarketContract.sol]: a neat trick.

File: contracts/oraclize/TestableMarketContractFactoryOraclize.sol [hunk 2]
```diff
@@ -78,4 +83,13 @@ contract TestableMarketContractFactoryOraclize is Ownable {
         require(registryAddress != address(0));
         marketContractRegistry = registryAddress;
     }
+
+    /*
+    currently adding this function pushes us over the edge for gas, for the time being we can leave it out.
```
Comment (cryptomental) [inline contracts/oraclize/TestableMarketContractFactoryOraclize.sol]: Do we really want to switch factories anyway?

**Comment by cryptomental** [comment]
I am wondering why Travis complains on those two lines not covered compared to master https://coveralls.io/builds/17315005/source?filename=contracts%2FMarketContract.sol#L261 , it would be good to double check why the error is not emitted anymore during cancelOrder.

File: contracts/oraclize/TestableMarketContractFactoryOraclize.sol (hunk 2 above)
Comment (pelsasser) [reply contracts/oraclize/TestableMarketContractFactoryOraclize.sol]: The only reason I can see to switch factories would be in production if we wanted to fix an issue in just the CollateralPoolFactory contract, we could and then hot swap the contracts.

**Comment by pelsasser** [comment]
@cryptomental that is a good question, will look into it. thank you for the review!

**Comment by pelsasser** [comment]
@cryptomental - I think this is an issue with coverage.  I updated the tests in https://github.com/MARKETProtocol/MARKETProtocol/pull/102/commits/004b583e672ee0ae5210593887481ab0adeee1cb to be sure that the specific error code was generated and magically coverage change.

I am going to merge.

@captnseagraves sorry for throwing you down a wild goose chase on this one!

**Pull request merged**